File: 서로소 집합.cpp
#include <iostream>
using namespace std;
int par[100];

void init(){
	for(int i=0;i<100;i++)par[i]=i;
}
int find(int u){
	return u==par[u] ? u : par[u]=find(par[u]); //Path Compression
}
void merge(int u,int v){
	u=find(u);v=find(v);
	if(u==v)return;
	par[u]=v; // without Union-By-Rank
}

int main(){
	init();
	merge(1,5);
	merge(2,5);
	merge(4,1);
	cout<<(find(5)==find(4))<<endl;
	cout<<(find(0)==find(1))<<endl;
}

File: 서로소 집합.py
par=[]
for i in range(100):
    par.append(i)

def find(u):
    if u==par[u]:
        return u
    par[u]=find(par[u]) # Path Compression, 경로 압축
    return par[u]

def merge(u,v):
    u,v=find(u),find(v)
    if u==v:
        return
    par[u]=v # without Union-By-Rank

merge(1,5)
merge(2,5)
merge(4,1)

print(find(5)==find(4))
print(find(0)==find(1))